src/components/organisms/OakCookieSettingsModal/OakCookieSettingsModal.tsx

```diff
@@ -0,0 +1,217 @@
+import React, { FormEvent, ReactNode } from "react";
+
+import {
+  OakAccordion,
+  OakCheckBox,
+  OakLink,
+  OakModal,
+  OakModalBody,
+  OakModalFooter,
+  OakModalProps,
+  OakPrimaryButton,
+  OakSecondaryButton,
+} from "@/components/molecules";
+import { OakBox, OakHeading, OakP, OakSpan, OakUL } from "@/components/atoms";
+
+type Party = {
+  name: ReactNode;
+  policyURL: string;
+};
+type Policy = {
+  /**
+   * Unique identifier for the policy.
+   */
+  id: string;
+  /**
+   * Label for the policy.
+   */
+  label: string;
+  /**
+   * Description of the policy. This should explain what the policy does and why it is needed.
+   */
+  description: ReactNode;
+  /**
+   * Whether the policy is strictly necessary for the site to function.
+   * If true, the policy will be enabled by default and cannot be disabled.
+   */
+  strictlyNecessary: boolean;
+  /**
+   * List of 3rd parties that the policy allows data to be shared with.
+   */
+  parties: Party[];
+};
+type ConsentState = "granted" | "denied";
+type Consents = {
+  [policyId: string]: ConsentState;
+};
```

Comment:
The intent here is that these align with https://github.com/oaknational/oak-consent-manager as much as is practical.